bindings/rust/extended/s2n-tls/src/cert_chain.rs

```diff
@@ -125,6 +157,36 @@ impl Builder {
         Ok(self)
     }
 
+    /// Associates an arbitrary application context with the CertificateChain to
+    /// be later retrieved via [`CertificateChain::application_context()`].
+    ///
+    /// This API will override an existing application context set on the Builder.
+    ///
+    /// Corresponds to [s2n_cert_chain_and_key_set_ctx].
+    pub fn set_application_context<T: Send + Sync + 'static>(
+        &mut self,
+        app_context: T,
+    ) -> Result<&mut Self, Error> {
+        let app_context = Box::new(app_context);
+        match self.cert_handle.internal_context_mut() {
+            // set_application_context was previously called, overwrite the existing value
+            Some(c) => c.context = app_context,
```

Comment:
This seems reasonable if you could get/set different contexts throughout the lifetime of the cert chain. But since this can only be called on the builder, won't overriding this context basically always be a mistake? Should this error?